startLuminoWithRIFComms.sh

```diff
@@ -0,0 +1,130 @@
+#!/bin/bash
+
+### Usage
+
+## This script assumes:
+
+# You have node 14.13.1 installed
+# You have ran npm install on the rif communications pub sub bootnode project
+# You have created the config files for the rif communications pub sub bootnode (i.e /config/nodeA and keys/nodeA)
+# You already know and configured the PORT in which rif comms node is going to run
+
+## Running
+
+# Run this script in the directory in which Lumino was git cloned
+# Specify the flags that are part of ARGUMENT_LIST, i.e
+#
+#. /startLuminoWithRIFComms.sh --RIF_COMMS_NODE_DIR="~/rsk/rif-communications-pubsub-bootnode" --KEYSTORE_PATH="/home/marcos/rsk/keystore" --TOKEN_NETWORK_REGISTRY="0x07e1CD1ea2123e3a0f624553761EaD1c1e150CC3" --SECRET_REGISTRY="0xfA8Bc06C815BD4C4bB641ccb00EBF9CB8BEB2d67" --ENDPOINT_REGISTRY="0xC14E9A67A87949bC131b9f22a4EB5cd9d9A6728e" --HUB_MODE=0 --RSK_ENDPOINT="http://localhost:4444" --LUMINO_API_ENDPOINT="http://localhost:5001" --NETWORK_ID=33 --RIF_COMMS_NODE_CONFIGURATION="nodeA" --RIF_COMMS_NODE_PORT=5013
+
+## Here some explanation of the parameters:
+
+## TOKEN_NETWORK_REGISTRY: The token network registry contract address to be used by the node.
+## SECRET_REGISTRY: The secret registry contract address to be used by the node.
+## ENDPOINT_REGISTRY: The endpoint registry contract address to be used by the node.
+## HUB_MODE: If set, the node will work as a hub
+## RSK_ENDPOINT: The RSK node endpoint URL
+## LUMINO_API_ENDPOINT: The expose url to use by the lumino node
+## NETWORK_ID: the blockchain network ID to connect to
+## RIF_COMMS_NODE_CONFIGURATION: config file of the rif comms node
+## RIF_COMMS_NODE_PORT: port in which rif comms node is going to start
+
+
+# The output log for rif comms node is on rif-communications-pubsub-bootnode under the commsNodeLog.txt name
+
+ARGUMENT_LIST=(
+    "RIF_COMMS_NODE_DIR"
+    "KEYSTORE_PATH"
+    "TOKEN_NETWORK_REGISTRY"
+    "SECRET_REGISTRY"
+    "ENDPOINT_REGISTRY"
+    "HUB_MODE"
+    "RSK_ENDPOINT"
+    "LUMINO_API_ENDPOINT"
+    "NETWORK_ID"
+    "RIF_COMMS_NODE_CONFIGURATION"
+    "RIF_COMMS_NODE_PORT"
+)
+
+
+# read arguments
+opts=$(getopt \
+    --longoptions "$(printf "%s:," "${ARGUMENT_LIST[@]}")" \
+    --name "$(basename "$0")" \
+    --options "" \
+    -- "$@"
+)
+
+eval set --$opts
+
+while [[ $# -gt 0 ]]; do
```

Comment:
If this are the same as the child script you can skip this and send the params to the child directly from this parent, something like this should be ok ```./startLumino.sh $@```, that will pass the same params from this script ot the child script and you don't need to validate them all here again.

Reply:
oh thats nice! i will take a look at this

Reply:
The thing is, that the specific variables to rif comms, for example: RIF_COMMS_NODE_PORT, must be validated on the parent script, and, since we are using $0, we are passing all the params to the child.

The child, doesnt recognize the RIF_COMMS_NODE_PORT flag... 

we must have a way to send a subset of $0

what do u think?

Reply:
Yeah i get that, you can use a slice in that case if you want, be careful because @0 doesn't do anything, i know you mean to put $@, anyway using the slice should work, here some examples:

```bash
A=( foo bar "a  b c" 42 )
B=("${A[@]:1:2}")
C=("${A[@]:1}")       # slice to the end of the array
echo "${B[@]}"        # bar a  b c
echo "${B[1]}"        # a  b c
echo "${C[@]}"        # bar a  b c 42
echo "${C[@]: -2:2}"  # a  b c 42 # The space before the - is necesssary

```

you can see it [here](https://stackoverflow.com/questions/1335815/how-to-slice-an-array-in-bash)